pym/bob/input.py

```diff
@@ -576,6 +577,62 @@ def isDeterministic(self):
     def hasJenkinsPlugin(self):
         return True
 
+    def callGit(self, workspacePath, *args):
+        cmdLine = ['git']
+        for arg in args:
+            cmdLine.append(arg)
+        try:
+            output = subprocess.check_output(cmdLine, cwd=os.path.join(os.getcwd(),workspacePath, self.__dir),
+                universal_newlines=True, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            raise BuildError("git error: '{}' '{}'".format(" ".join(cmdLine), e.output))
+        return output
+
+    def status(self, workspacePath, dir, verbose):
+        if not os.path.exists(os.path.join(os.getcwd(), workspacePath, self.__dir)):
+            return 'error'
+
+        ret = 0
+        if self.__commit:
+            output = self.callGit(workspacePath, 'rev-parse', 'HEAD').rstrip()
+            if output != self.__commit:
+                ret += 1
+                if verbose:
+                    print(colorize("    ! should have commitId {} but commitID is {}".format(self.__commit, output), "33"))
```

Comment:
this information is given in build.py - showStatus (arround line 1100). 
```python
for (dir, hash) in scm.getDirectories().items():
    print(colorize("   STATUS {}".format(os.path.join(package.getCheckoutStep().getWorkspacePath(), dir)), "32"))
    scm.status(package.getCheckoutStep().getWorkspacePath(), dir, True)
```
I would move this to input.py and print this STATUS message only in unclean case for verbose=1? This would remove worthless output and focus on the unclean dirs.